e2e/internal/upgrade/image.go

```diff
@@ -0,0 +1,132 @@
+//go:build e2e
+
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package upgrade
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/edgelesssys/constellation/v2/internal/attestation/measurements"
+	"github.com/edgelesssys/constellation/v2/internal/cloud/cloudprovider"
+	"github.com/edgelesssys/constellation/v2/internal/versionsapi"
+	"github.com/edgelesssys/constellation/v2/internal/versionsapi/fetcher"
+)
+
+type measurement struct {
+	WarnOnly bool   `json:"warnOnly"`
+	Expected string `json:"expected"`
+}
+
+type imageMeasurements struct {
+	Cmdline       string `json:"cmdline"`
+	CmdlineSHA256 string `json:"cmdline-sha256"`
+	EFIstages     []struct {
+		Name   string `json:"name"`
+		SHA256 string `json:"sha256"`
+	} `json:"efistages"`
+	InitrdSHA256 string                 `json:"initrd-sha256"`
+	Measurements map[string]measurement `json:"measurements"`
+}
+
+type upgradeInfo struct {
+	measurements measurements.M
+	shortPath    string
+	wantImage    string
+}
+
+func fetchUpgradeInfo(csp cloudprovider.Provider, toImage string) (upgradeInfo, error) {
+	info := upgradeInfo{
+		measurements: make(measurements.M),
+		shortPath:    toImage,
+	}
+	versionsClient := fetcher.NewFetcher()
+
+	ver, err := versionsapi.NewVersionFromShortPath(toImage, versionsapi.VersionKindImage)
+	if err != nil {
+		return upgradeInfo{}, err
+	}
+
+	measurementsURL, _, err := ver.MeasurementURL(csp)
```

Comment:
```suggestion
	measurementsURL, _, err := versionsapi.MeasurementURL(ver, csp)
```
`versionsapi.Version` does not have a method `MeasurementURL`